apps/design/backend/src/app.ts

```diff
@@ -828,6 +846,45 @@ export function buildApi({ auth0, logger, workspace, translator }: AppContext) {
         ]);
       }
     },
+
+    processQrCodeReport({
+      payload,
+    }: {
+      payload: string;
+      // TODO (#7150): Verify the signature and receive a machine public cert in the request as well
+      signature: string;
+    }): Result<ResultsReportInfo, ResultsReportingError> {
+      const messageParts = payload.split('//');
+      const [version, header, message] = messageParts;
+
+      if (version !== '1' || header !== 'qr1') {
+        return err('invalid-payload');
+      }
+
+      const [ballotHash, machineId, isLive, secondsSince1970, tallyB64] =
+        message.split('.');
+      const timestampNum = safeParseNumber(secondsSince1970);
+      if (timestampNum.isErr()) {
+        return err('invalid-payload');
+      }
+
+      const signedTimestamp = new Date(timestampNum.ok() * 1000);
+      let tally: CompressedTally;
+      try {
+        const tallyJson = Buffer.from(tallyB64, 'base64').toString('ascii');
+        tally = JSON.parse(tallyJson) as CompressedTally;
+        // TODO (#7151): Load the election for the given ballot hash, save the results report.
+      } catch (error) {
+        return err('invalid-payload');
+      }
+      return ok({
+        ballotHash,
+        machineId,
+        isLive: isLive === '1',
+        signedTimestamp,
+        tally,
+      });
+    },
```

Comment:
It might be worth exploring making a separate grout API for this method since it has different middleware needs.

Reply:
oh interesting I didn't think of that, will see if thats easy to set up